examples/crystal.py

```diff
@@ -289,28 +294,42 @@ def simulate_crystal_with_gaussian_beam(
         brillouin_grid_shape, primitive_lattice_vectors
     )
     in_plane_wavevector += basis.plane_wave_in_plane_wavevector(
-        wavelength=jnp.asarray(wavelength),
+        wavelength=jnp.asarray(wavelengths),
         polar_angle=jnp.asarray(polar_angle),
         azimuthal_angle=jnp.asarray(azimuthal_angle),
         permittivity=jnp.asarray(permittivity_ambient),
     )
+
+    assert in_plane_wavevector.shape[0] == wavelengths.size
+    assert in_plane_wavevector.shape[1] == brillouin_grid_shape[0]
+    assert in_plane_wavevector.shape[2] == brillouin_grid_shape[1]
     assert in_plane_wavevector.shape[-1] == 2
-    assert in_plane_wavevector.ndim == 3
+    assert in_plane_wavevector.ndim == 4
 
     eigensolve = functools.partial(
         fmm.eigensolve_isotropic_media,
-        wavelength=jnp.asarray(wavelength),
+        wavelength=jnp.asarray(wavelengths),
         in_plane_wavevector=in_plane_wavevector,
         primitive_lattice_vectors=primitive_lattice_vectors,
         expansion=expansion,
         formulation=fmm.Formulation.FFT,
     )
 
+    def _reshape_permittivity(arr: jnp.ndarray):
```

Comment:
Is this needed? The automatic broadcasting might already take care of this.

Reply:
Fixed, thanks.